apps/system/js/notifications.js

```diff
@@ -408,9 +408,13 @@ var NotificationScreen = {
     // timesout, so we skip updating it here, by passing a skip bool
     this.addUnreadNotification(detail.id, true);
 
+    var cIndex = this.availableNotification.indexOf(detail.appName);
+    if (cIndex === -1) {
+    this.availableNotification.push(detail.appName);
```

Comment:
I notice that there is no code that ever removes notifications from this array. Is that correct? Does the NotificationsScreen object get torn down and rebuilt each time the lock screen is invoked?  If not, then I'm pretty sure you'll have to manage the array contents more actively.